content/07.architecture-and-representation.md

```diff
@@ -1 +1,3 @@
-## Tip 5: Choose an appropriate neural network architecture and data representation {#architecture}
+## Tip 5: Choose an appropriate data representation and neural network architecture {#architecture}
+Unfortunately, choosing how to represent your data and design your architecture is closer to an art than a science.
+With that being said, there are some general principles that are useful to follow when experimenting.
```

Comment:
This is a nice start! Here a suggestion to extend this a bit:

Unfortunately, choosing how to represent your data and design your architecture is closer to an art than a science. While certain best practices have been established by the research community {cite see below}, architecture design choices remain largely problem-specific and vastly empirical efforts requiring extensive experimentation. Furthermore, as deep learning is a fast moving field, many recommendations are often short-lived and frequently replaced by newer insights supported by recent empirical results. This is further complicated by the fact that many recommendations are very problem- and dataset-specific and do generalize well across different tasks. With that being said, there are some general principles that are useful to follow when experimenting.

- I recommend citing the collection "Neural Networks: Tricks of the Trade" to which many established authors contributed their insights: https://doi.org/10.1007/978-3-642-35289-8

Reply:
Well said!

Did you mean to write:
> very problem- and dataset-specific and do _**not**_ generalize

Reply:
Oh yes, definitely meant **not** generalize here :)

Reply:
Agreed. I like @rasbt 's extension.